dinstaller-lib/src/install_settings.rs

```diff
@@ -37,19 +101,39 @@ impl Settings for InstallSettings {
     fn set(&mut self, attr: &str, value: SettingValue) -> Result<(), &'static str> {
         if let Some((ns, id)) = attr.split_once('.') {
             match ns {
-                "software" => self.software.set(id, value)?,
-                "user" => self.user.set(id, value)?,
-                "storage" => self.storage.set(id, value)?,
+                "software" => {
+                    let software = self.software.get_or_insert(Default::default());
+                    software.set(id, value)?
+                }
+                "user" => {
+                    let user = self.user.get_or_insert(Default::default());
+                    user.set(id, value)?
+                }
+                "storage" => {
+                    let storage = self.storage.get_or_insert(Default::default());
+                    storage.set(id, value)?
+                }
                 _ => return Err("unknown attribute"),
             }
         }
         Ok(())
     }
 
-    fn merge(&mut self, other: Self) {
-        self.software.merge(other.software);
-        self.user.merge(other.user);
-        self.storage.merge(other.storage);
+    fn merge(&mut self, other: &Self) {
+        if let Some(other_software) = &other.software {
+            let software = self.software.get_or_insert(Default::default());
+            software.merge(other_software);
+        }
+
+        if let Some(other_user) = &other.user {
+            let user = self.user.get_or_insert(Default::default());
+            user.merge(other_user);
+        }
+
+        if let Some(other_storage) = &other.storage {
+            let storage = self.storage.get_or_insert(Default::default());
+            storage.merge(other_storage);
```

Comment:
I start feeling that we have a lot of repetition here, not mentioning when we want to add new scope, we need to add it to a lot of placces

Reply:
Yes, sure. I plan to use a macro to reduce all this repetition, but I want a clear view of the needed code first.